BM64.py

```diff
@@ -298,81 +592,137 @@ def get_analyzer_frame(self, start_time, end_time, rx_channel):
             return AnalyzerFrame('bm64', start_time, end_time, {
                 'packet_type': "BM64 Command",
-                'opcode': event_code,
                 'operation': event_str,
-                'packet_length': length,
             })
         else:
-            if event_code in BM64_EVENT_DESC:
-                event_str = BM64_EVENT_DESC[event_code]
-            return AnalyzerFrame('bm64', start_time, end_time, {
-                'packet_type': "BM64 Event",
-                'operation': event_str,
-                'packet_length': length,
+            data = self._data
+            data_str = ''
+            event_str = BM64_EVENT_DESC[event_code]
+            if event_code == 0x00:
+                data_str = BM64_COMMAND_DESC[data[0]]
+                if data[1] == 0x00:
+                    data_str += ' - Command complete: BTM can handle this command.'
+                elif data[1] == 0x01:
+                    data_str += ' - Command disallow:: BTM cannot handle this command.'
+                elif data[1] == 0x02:
+                    data_str += ' - Unknown command'
+                elif data[1] == 0x03:
+                    data_str += ' - Parameters error'
+                elif data[1] == 0x04:
+                    data_str += ' - BTM is busy'
+                elif data[1] == 0x05:
+                    data_str += ' - BTM memory is full'
+            elif event_code == 0x01:
+                data_str = BTMStatus(data).get_string()
+            elif event_code == 0x2d:
+                data_str = ReportTypeCodec(data).get_string()
+            elif event_code == 0x30:
+                data_str = "Initialization complete"
+            elif event_code == 0x32:
+                data_str = LESignalingEvent(data).get_string()
+            elif event_code == 0x39:
+                data_str = ReportLEGATTEvent(data).get_string()
+            else:
+                data_str = 'Not implemented event code: ' + \
+                    str(hex(event_code))
+
+            return AnalyzerFrame('BM64 Event', start_time, end_time, {
+                'packet_type': "BM64 Event:",
+                'event': event_str,
+                'data': data_str,
             })
 
+
 # Define all the supported packets by their type
 PACKETS = {
-    0x00: BM64PacketWakeup, # A bm64 package with wakeup b'0x00
+    0x00: BM64PacketWakeup,  # A bm64 package with wakeup b'0x00
     0x01: HCICommandPacket,
     0x02: HCIISDAPFlashPacket,
     0x04: HCIEventPacket,
     0xAA: BM64Packet,
 }
 
-# Collect all the result types
-RESULT_TYPES = {}
-for p in PACKETS.values():
-    RESULT_TYPES.update({k: {'format': v} for k, v in p.RESULT_TYPES.items()})
 
 # High level analyzers must subclass the HighLevelAnalyzer class.
+
+
 class Hla(HighLevelAnalyzer):
+
+    result_types = {
+        'bm64': {
+            'format': '{{data.packet_type}} - {{data.operation}} {{data.data}}'
+        },
+        'BM64 Event': {

```

Comment:
keep case style as the others? ```bm64_event```

Reply:
Also, is this referred to anywhere? It seems like the bm64 events just use the bm64 result type.